submission/programs/adae.R

```diff
@@ -0,0 +1,350 @@
+  derive_vars_merged(
+    dataset_add = adsl,
+    new_vars = adsl_vars,
+    by = vars(STUDYID, USUBJID) 
+  ) %>%
+  
+  #------------------------------#
+  # Set TRTA and TRTAN from ADSL #
+  #------------------------------#
+  rename(TRTA = TRT01A,
+         TRTAN = TRT01AN) %>% 
+  # -----------------------------#
+  # Derive analysis start time --#
+  # -----------------------------#
+  derive_vars_dtm(
+    dtc = AESTDTC,
+    new_vars_prefix = "AST",
+    highest_imputation = "D" 
+  )   %>% 
+
+  # -----------------------------#
+  # Derive analysis end time     #
+  # -----------------------------#
+  derive_vars_dtm(
+    dtc = AEENDTC,
+    new_vars_prefix = "AEN",
+    highest_imputation = "M",
+    date_imputation = "last",
+    time_imputation = "last",
+    # max_dates = vars(DTHDT, EOSDT)
+    max_dates = NULL
+  )%>%
+  
+  
+  # -----------------------------------#
+  # Derive analysis start & end dates  #
+  # -----------------------------------#
+  derive_vars_dtm_to_dt(vars(ASTDTM , AENDTM )) %>% 
+  mutate (
+    TRTSDTM = as_datetime(TRTSDT)  , 
+    astdt_m = as_datetime(as.Date(AESTDTC))  , 
+    aendt_m = as_datetime(as.Date(AEENDTC))  
+  ) %>% 
+
+  # ----------------------------#
+  # Analysis Start Relative Day #  
+  # Analysis End Relative Day   #
+  # ----------------------------#
+  derive_vars_dy(
+    reference_date = TRTSDTM,
+    source_vars = vars(TRTSDTM, ASTDTM , AENDTM)
+  ) %>%
+  
+  derive_vars_duration( 
+                       new_var = ADURN,
+                       new_var_unit = ADURU,
+                       start_date = astdt_m,
+                       end_date = aendt_m,
+                       out_unit = "days"
+  ) %>% 
+  mutate (ADURU = str_replace(ADURU , "DAYS" , "DAY")) %>% 
+  # ---------------------------------#
+  # Treatment Emergent Analysis flag #
+  # ---------------------------------#
+  
+  ### NOTE : These 5 lines below are code from admiral{} that may need to be udpated.
+  # derive_var_trtemfl(
+  #   new_var = TRTEMFL, start_date = ASTDT , end_date = AENDT ,
+  #   trt_start_date = TRTSDT , trt_end_date = NULL, end_window = NULL,
+  #   ignore_time_for_trt_end = TRUE, initial_intensity = NULL,  intensity = NULL
+  # )
+
+  mutate(TRTEMFL = if_else(ASTDT >= TRTSDT , "Y" , NA_character_) ) %>%
+  
+  #---------------------------------------------------------------------#
+  # AOCCFL - 1st Occurrence of Any AE Flag                              #
+  #---------------------------------------------------------------------#
+  # Programming Notes:                                                  #
+  # Subset to TRTEMFL='Y' and sort by Subject (USUBJID),                #
+  # Start Date (ASTDT), and Sequence Number (AESEQ) and                 #
+  # flag the first record (set AOCCFL=’Y’) within each Subject          #
+  # Not matching on 01-701-1118  as TRTEMFL = "Y"   as ASTDT is missing #
+  #---------------------------------------------------------------------#  
```

Comment:
Please remove these kind of comments which are copied from the specs.

Reply:
Removed.

Reply:
Please review and resolve to ensure you do not see it on your end.

Reply:
@SHAESEN2 or @thomas-neitmann If ok, please `Resolve conversation`.